accounts/external/backend.go

```diff
@@ -208,14 +208,36 @@ func (api *ExternalSigner) SignTx(account accounts.Account, tx *types.Transactio
 		t := common.NewMixedcaseAddress(*tx.To())
 		to = &t
 	}
-	args := &core.SendTxArgs{
-		Data:     &data,
-		Nonce:    hexutil.Uint64(tx.Nonce()),
-		Value:    hexutil.Big(*tx.Value()),
-		Gas:      hexutil.Uint64(tx.Gas()),
-		GasPrice: hexutil.Big(*tx.GasPrice()),
-		To:       to,
-		From:     common.NewMixedcaseAddress(account.Address),
+	args := &apitypes.SendTxArgs{
+		Data:  &data,
+		Nonce: hexutil.Uint64(tx.Nonce()),
+		Value: hexutil.Big(*tx.Value()),
+		Gas:   hexutil.Uint64(tx.Gas()),
+		To:    to,
+		From:  common.NewMixedcaseAddress(account.Address),
+	}
+	switch tx.Type() {
+	case types.LegacyTxType, types.AccessListTxType:
+		args.GasPrice = (*hexutil.Big)(tx.GasPrice())
+	case types.DynamicFeeTxType:
+		args.MaxFeePerGas = (*hexutil.Big)(tx.GasFeeCap())
+		args.MaxPriorityFeePerGas = (*hexutil.Big)(tx.GasTipCap())
+	default:
+		return nil, fmt.Errorf("Unsupported tx type %d", tx.Type())
+	}
+	// We should request the default chain id that we're operating with
+	// (the chain we're executing on)
+	if chainID != nil {
```

Comment:
if chainId is nil, were do we get the chain id to actually sign? Remember that we enforce EIP155 so signing without chainId is not an option

Reply:
I think this check is because someone using the legacyTx can (and should) send the chainId as a paramenter, but if it is using the new Tx, it will be asked from the `tx` instead of the parameter. This non-legacy way, will return a non nil/zero chainId.
Then, the 
```
	if err := api.client.Call(&res, "account_signTransaction", args); err != nil {
```
will return an error if the chainId is not the same as the signer's chainId
We can add an error in this first function (instead of failing after), but I should keep this function as a Legacy/new unifier, and the other with the actual requirements for a valid tx